src/vaadin-upload.html

```diff
@@ -527,6 +527,7 @@
 
           const ini = Date.now();
           const xhr = file.xhr = this._createXhr(file);
+          const index = this.files.indexOf(file);
```

Comment:
Introducing `this.files.indexOf(file)` has the implication of file always being stored in `this.files` 🤔 

But  technically the user could have `<input type="file">` somewhere outside of `<vaadin-upload>` and only use public `uploadFiles` API to trigger uploading (somewhat similar use case described #181)